### PR TITLE
LPS-89497

### PR DESCRIPTION
**Pull request by yuhai**

### CODE REVIEW

**Comment by yuhai** [comment]
ci:test:sf

**Comment by yuhai** [comment]
ci:test:relevant

**Comment by dantewang** [comment]
Fetched

File: modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/upgrade/test/ConfigurationUpgradeStepFactoryTest.java [hunk 1]
```diff
@@ -122,18 +124,18 @@ private void _testUpgradeConfig(
...
-							"felix.fileinstall.filename", uri.toString()));
 				}
 				else {
 					configuration = _configurationAdmin.getConfiguration(
 						oldPid);
 				}
+
+				URI uri = oldConfigFile.toURI();
+
+				ConfigurationTestUtil.saveConfiguration(
```
Comment (dantewang) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-test/src/testIntegration/java/com/liferay/portal/configuration/persistence/upgrade/test/ConfigurationUpgradeStepFactoryTest.java]: @yuhai The test case still doesn't handle the situation that the configuration doesn't have this property.

File: modules/apps/static/portal-configuration/portal-configuration-persistence-impl/src/main/java/com/liferay/portal/configuration/persistence/internal/upgrade/ConfigurationUpgradeStepFactoryImpl.java [hunk 2]
```diff
@@ -56,6 +56,15 @@ public UpgradeStep createUpgradeStep(String oldPid, String newPid) {
 
 					dictionary.put("service.pid", newPid);
 
+					String fileName = dictionary.get(
+						"felix.fileinstall.filename");
+
+					if (fileName != null) {
```
Comment (dantewang) [inline modules/apps/static/portal-configuration/portal-configuration-persistence-impl/src/main/java/com/liferay/portal/configuration/persistence/internal/upgrade/ConfigurationUpgradeStepFactoryImpl.java]: @yuhai Can you say again when will this be set for non-factory configuration?

Comment (yuhai) [reply modules/apps/static/portal-configuration/portal-configuration-persistence-impl/src/main/java/com/liferay/portal/configuration/persistence/internal/upgrade/ConfigurationUpgradeStepFactoryImpl.java]: When place the config file in osgi/configs/, the property will been added by org-apache-felix-fileinstall.

**Pull request closed without merging**